### ISSUE TITLE
"Skipped parameter" warnings in jenkins server log

### ISSUE THREAD

**Issue by sxa** [adoptium/infrastructure#2774]
These aren't new but were discovered while working on #2108

There are a lot of warnings in the jenkins log which we should look at clearing up in the interests of avoid the risk of being unable to see the wood for the trees. All are related to `NODE_LABEL`'s use in various places.

Presumably the parameters are being passed in from the upstream job somewhere but they are not defined on the callee so are superfluous. Can we wasily stop 'rogue' parameters being passed in or should we set the `-D` mentioned in the error?

90 WARNING	hudson.model.ParametersAction#filter: Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/release/create_installer_mac`. Set `-Dhudson.model.ParametersAction.keepUndefinedParameters=true` to allow undefined parameters to be injected as environment variables or `-Dhudson.model.ParametersAction.safeParameters=[comma-separated list]` to whitelist specific parameter names, even though it represents a security breach or `-Dhudson.model.ParametersAction.keepUndefinedParameters=false` to no longer show this message.

**Comment by sxa** [comment]
FYI @andrew-m-leonard  @smlambert (Shelley I sent you a log from this a while back and I think there may be fewer such messages now but unclear if that's my memory or if it was a slightly different error!

**Comment by sxa** [comment]
Also tagging @sophia-guo since she was involved in the discussions on the community call earlier today :-)

**Comment by sophia-guo** [comment]
According to the discussion if jenkins jobs are triggered with undefined parameters there will be some `warnings` in jenkins server log? If that is the case those warnings can be avoided by re-generating aqa-tests or smoketests jobs triggered by upstream. 

Initially when new features was added there was a period that aqa-tests or smoke tests jobs were triggered with undefined parameters. For example, USE_TESTENV_PROPERTIES, the reason is testJobTemplate was not updated accordingly as it didn't affect the functionalities. I have double checked all parameters of aqa-tests and smoke tests, all of them have been updated in testJobTemplate. That means if there is still those warning message related with aqa-tests or smoketests regenerating those jobs should fix it. Note autoGen is enabled with aqa-tests and disabled with smoketests.

**Comment by sxa** [comment]
Thanks for checking @sophia-guo! It looks like the remaining ones are mostly on the build pipelines so the test jobs may now be all sorted. I've just done a check on the recent logs and these are the remaining warnings - interestingly only the (new) jdk19u pipelines are showing up here @andrew-m-leonard:

```
jenkins@jenkins-hetzner-ubuntu2004:~$ gzip -cd /var/log/jenkins/jenkins.log.?.gz | grep Skipped.*undefined /var/log/jenkins/jenkins.log | sed -e 's/.*Skipped/Skipped/g' -e 's/. Set.*/./g' | sort | uniq -c
    282 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk19u/jdk19u-aix-ppc64-temurin`.
    291 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk19u/jdk19u-alpine-linux-aarch64-temurin`.
    306 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk19u/jdk19u-alpine-linux-x64-temurin`.
    291 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk19u/jdk19u-linux-aarch64-temurin`.
    286 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk19u/jdk19u-linux-arm-temurin`.
    278 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk19u/jdk19u-linux-ppc64le-temurin`.
    270 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk19u/jdk19u-linux-s390x-temurin`.
    294 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk19u/jdk19u-linux-x64-temurin`.
    485 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk19u/jdk19u-mac-aarch64-temurin`.
    488 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk19u/jdk19u-mac-x64-temurin`.
    432 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk19u/jdk19u-windows-x64-temurin`.
    447 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk19u/jdk19u-windows-x86-32-temurin`.
     36 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/release/create_installer_mac`.
     72 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/release/sign_installer`.
    240 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/release/sign_temurin_gpg`.
jenkins@jenkins-hetzner-ubuntu2004:~$ 
```

**Comment by andrew-m-leonard** [comment]
@sxa I don't understand the above, none of the build jobs have a Parameter called NODE_LABEL, they only have 4 paramaters:
- BUILD_CONFIGURATION
- USER_REMOTE_CONFIGS
- DEFAULTS_JSON
- ADOPT_DEFAULTS_JSON

**Comment by sxa** [comment]
That sounds consistent with the warning message - "it is undefined on <job>" but presumably something is trying to start them using those extra, unusable, parameters.

**Comment by sxa** [comment]
Had a look at the log for something else today. We've still got:
- `USE_TESTENV_PROPERTIES` on most of the test jobs giving the warning
- `RELEASE_TAG` showing ONLY on Test_openjdk11_j9_sanity.openjdk_x86-64_linux in the most recenr log. Not sure why that might be special.
- `NODE_LABEL` on most of the build jobs
- `GENERATE_JOBS` on most of the test jobs
- `DYNAMIC_COMPILE` on most o the test jobs.

**Comment by smlambert** [comment]
Regenerated the following test jobs:

JDK 11,17,19 hotspot/platforms
https://ci.adoptopenjdk.net/view/Test_grinder/job/Test_Job_Auto_Gen/864/ (functional - all levels)
https://ci.adoptopenjdk.net/view/Test_grinder/job/Test_Job_Auto_Gen/865/ (perf,system,openjdk - sanity,extended)

JDK 8 hotspot/platforms
https://ci.adoptopenjdk.net/view/Test_grinder/job/Test_Job_Auto_Gen/866/ (functional - all levels)
https://ci.adoptopenjdk.net/view/Test_grinder/job/Test_Job_Auto_Gen/867/ (perf,system,openjdk - sanity,extended)

**Comment by sxa** [comment]
@andrew-m-leonard After merging https://github.com/adoptium/ci-jenkins-pipelines/pull/668 we've still got a few remaining occurrences on the PR tester jobs which will need addressing:
```
    244 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts-pr-tester/build-test/jobs/jdk11u/jdk11u-linux-aarch64-temurin` (#114)
    373 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts-pr-tester/build-test/jobs/jdk11u/jdk11u-linux-x64-temurin` (#114)
    347 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts-pr-tester/build-test/jobs/jdk11u/jdk11u-mac-x64-temurin` (#115)
    174 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts-pr-tester/build-test/jobs/jdk11u/jdk11u-windows-x64-temurin` (#116)
    244 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts-pr-tester/build-test/jobs/jdk17u/jdk17u-linux-aarch64-temurin` (#109)
    373 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts-pr-tester/build-test/jobs/jdk17u/jdk17u-linux-x64-temurin` (#125)
    190 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts-pr-tester/build-test/jobs/jdk17u/jdk17u-mac-x64-temurin` (#109)
     94 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts-pr-tester/build-test/jobs/jdk17u/jdk17u-windows-x64-temurin` (#108)
    244 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts-pr-tester/build-test/jobs/jdk20/jdk20-linux-aarch64-temurin` (#8)
    373 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts-pr-tester/build-test/jobs/jdk20/jdk20-linux-x64-temurin` (#8)
    347 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts-pr-tester/build-test/jobs/jdk20/jdk20-mac-x64-temurin` (#8)
    174 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts-pr-tester/build-test/jobs/jdk20/jdk20-windows-x64-temurin` (#8)
    244 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts-pr-tester/build-test/jobs/jdk8u/jdk8u-linux-aarch64-temurin` (#105)
    363 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts-pr-tester/build-test/jobs/jdk8u/jdk8u-linux-x64-temurin` (#105)
    190 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts-pr-tester/build-test/jobs/jdk8u/jdk8u-mac-x64-temurin` (#105)
    382 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts-pr-tester/build-test/jobs/jdk8u/jdk8u-windows-x64-temurin` (#106)
```

**Comment by sxa** [comment]
Current status:
- Bisheng, Dragonwell and OpenJ9 build SmokeTest jobs have the issue with the `DYNAMIC_COMPILE` parameter
- In test jobs, there are issues with `GENERATE_JOBS`, `DYNAMIC_COMPILE` and `USE_TESTENV_PROPERTIES` and some others. Log of occurences today: [file.txt.gz](https://github.com/adoptium/infrastructure/files/11983866/file.txt.gz)

**Comment by sxa** [comment]
@smlambert @sophia-guo I just checked the logs today and most of the remaining ones seem to be from a small number of jobs - `Grinder_testList_0` and some of the JDK17 x64_windows ones. Since it only affects those can they be easily fixed? I presume a regen or somethign shoudl resolve this (can the grinder testlist one just be deleted if it will get regenerated if someone runs a grinder with Dynamic=Parallel?)

Occurences | parameter | job
---|---|---
704 | NON_AQA_TEST_REPOS|Grinder_testList_0
704 | PLATFORM_AND_MACHINE|Grinder_testList_0
704 | RERUN_ITERATIONS|Grinder_testList_0
616 | SETUP_JCK_RUN|Test_openjdk17_hs_extended.functional_x86-64_windows
910 | SETUP_JCK_RUN|Test_openjdk17_hs_extended.perf_x86-64_windows
612 | SETUP_JCK_RUN|Test_openjdk17_hs_extended.system_x86-64_windows
612 | SETUP_JCK_RUN|Test_openjdk17_hs_sanity.functional_x86-64_windows
731 | SETUP_JCK_RUN|Test_openjdk17_hs_sanity.openjdk_x86-64_windows
221 | SETUP_JCK_RUN|Test_openjdk17_hs_sanity.perf_x86-64_windows
876 | SETUP_JCK_RUN|Test_openjdk17_hs_sanity.system_x86-64_windows
703 | SETUP_JCK_RUN|Test_openjdk17_hs_special.functional_x86-64_windows
704 | TEST_IMAGES_REQUIRED|Grinder_testList_0

**Comment by smlambert** [comment]
Correct, we can delete that job, as it will be regenerated automatically next time someone runs a parallel grinder.

**Comment by sxa** [comment]
Thanks - I was hoping it would be that simple :-) I'll try and remember to check the logs again tomorrow

**Comment by sxa** [comment]
Still quite a lot of these:
- 5213 from openjdk17 Alpine jobs
- 5243 in openjdk17 arm32 jobs

The others are:
```
jenkins@jenkins-hetzner-ubuntu2004:/var/log/jenkins$ grep -i skipped.parameter jenkins.log | grep -v openjdk17.*alpine | grep -v openjdk17.*arm_ | cut -d\` -f4 | sort | uniq -c
      1 build-scripts/jobs/jdk11u/jdk11u-aix-ppc64-hotspot
      2 build-scripts/jobs/jdk11u/jdk11u-linux-aarch64-bisheng_SmokeTests
      2 build-scripts/jobs/jdk11u/jdk11u-linux-aarch64-dragonwell_SmokeTests
    204 build-scripts/jobs/jdk11u/jdk11u-linux-aarch64-openj9_SmokeTests
    229 build-scripts/jobs/jdk17u/jdk17u-linux-aarch64-openj9_SmokeTests
    229 build-scripts/jobs/jdk17u/jdk17u-linux-x64-openj9_SmokeTests
      4 Test_openjdk11_bisheng_extended.functional_aarch64_linux
      4 Test_openjdk11_bisheng_extended.system_aarch64_linux
      4 Test_openjdk11_bisheng_sanity.functional_aarch64_linux
    174 Test_openjdk11_bisheng_sanity.openjdk_aarch64_linux
      4 Test_openjdk11_bisheng_sanity.perf_aarch64_linux
    102 Test_openjdk11_bisheng_sanity.system_aarch64_linux
      6 Test_openjdk11_dragonwell_extended.functional_aarch64_linux
      6 Test_openjdk11_dragonwell_extended.system_aarch64_linux
      6 Test_openjdk11_dragonwell_sanity.functional_aarch64_linux
    234 Test_openjdk11_dragonwell_sanity.openjdk_aarch64_linux
      6 Test_openjdk11_dragonwell_sanity.perf_aarch64_linux
      6 Test_openjdk11_dragonwell_sanity.system_aarch64_linux
      6 Test_openjdk11_hs_extended.functional_arm_linux
      6 Test_openjdk11_hs_extended.system_arm_linux
      6 Test_openjdk11_hs_sanity.functional_arm_linux
    618 Test_openjdk11_hs_sanity.perf_arm_linux
      6 Test_openjdk11_hs_sanity.system_arm_linux
```
Most popular parameters causing issues are:
- DYNAMIC_COMPILE
- USE_TESTENV_PROPERTIES
- GENERATE_JOBS
- NODE_LABEL (Only one occurrence in today's logs)

**Comment by smlambert** [comment]
thanks @sophia-guo for https://github.com/adoptium/ci-jenkins-pipelines/pull/883 which will regen test jobs regularly and thus relieving the need to manually run regen, and keep the test jobs up-to-date, therefore reducing skipped warnings from those jobs
- I am going to remove my name as assignee to this ongoing issue

**Comment by sxa** [comment]
> which will regen test jobs regularly and thus relieving the need to manually run regen, and keep the test jobs up-to-date, therefore reducing skipped warnings from those jobs

My expectation from last weeks' conversations was that this would have caused a regeneration earlier this week so they would be in sync. Is there a chance that didn't happen? I don't know exactly where the regeneration takes place but it would be good to verify if it did happen, and understand if there is still an underlying issue that needs to be addressed.

**Comment by sophia-guo** [comment]
Weekly pipeline with parameter releaseType=Weekly will trigger the regen https://ci.adoptium.net/job/build-scripts/job/weekly-openjdk11-pipeline/156/parameters/. The code is in https://github.com/adoptium/ci-jenkins-pipelines/pull/883/files.

The regen didn't happen as the typo. The fix PR https://github.com/adoptium/ci-jenkins-pipelines/pull/883/files

**Comment by sxa** [comment]
Just realised I have this in a draft in my browser but hadn't sent it. Based on the runs this week:

We are down to one instance in the build jobs:
```
      1 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk11u/jdk11u-linux-arm-hotspot` (#933).
```
And the rest are all in test jobs, including some in the new `_rerun` jobs, which might indicate that the generation is still not quite doing what it should. Here is a complete list of what is outstanding (prefixed by a number showing the number of instances in the latest log): 
[jenkins-skipped-parameters.gz](https://github.com/adoptium/infrastructure/files/14099809/jenkins-skipped-parameters.gz)
A lot of them seem to be Bisheng/Dragonwell and the riscv64 ones - can we get those explicitly regenerated?

**Comment by smlambert** [comment]
I was going to propose that we stop building and testing the jobs that no one is caring for, as part of our cleanup prior to Trestle.  This includes openj9, bisheng, dragonwell, corretto, etc.

**Comment by sxa** [comment]
> I was going to propose that we stop building and testing the jobs that no one is caring for, as part of our cleanup prior to Trestle. This includes openj9, bisheng, dragonwell, corretto, etc.

I think I've proposed that in the past but there was a desire to keep them to demonstrate that they still worked. It would be nice in the short term to trigger a regen if possible though (I'm guessing they aren't in the weekly pipelines at present, so likely won't be in the 'trigger on tag' builds that we replace those with) and persue potential de-scoping of other variants under a separate issue. We could even have a "build grinder" job that people can trigger if desired.

My gut feel is that it's "nice" to try and verify that we haven't broken the ability to build those other projects with our scripts, but we could raise it with the working group, since if test is not interested then there may be no takers. It seems that no-one in build is actively looking at the results of those pipelines at present.

**Comment by sxa** [comment]
@smlambert @sophia-guo Looking at one of the logs for Bisheng it has:
```
17:13:43  Scheduling project: [Test_Job_Auto_Gen](https://ci.adoptium.net/job/Test_Job_Auto_Gen/)
[Pipeline] build (Building Test_Job_Auto_Gen)
17:13:43  Scheduling project: [Test_Job_Auto_Gen](https://ci.adoptium.net/job/Test_Job_Auto_Gen/)
[Pipeline] build (Building Test_Job_Auto_Gen)
17:13:43  Scheduling project: [Test_Job_Auto_Gen](https://ci.adoptium.net/job/Test_Job_Auto_Gen/)
17:13:45  Starting building: [Test_openjdk11_bisheng_sanity.openjdk_x86-64_linux #301](https://ci.adoptium.net/job/Test_openjdk11_bisheng_sanity.openjdk_x86-64_linux/301/)
17:13:45  Starting building: [Test_openjdk11_bisheng_sanity.system_x86-64_linux #301](https://ci.adoptium.net/job/Test_openjdk11_bisheng_sanity.system_x86-64_linux/301/)
17:13:45  Starting building: [Test_openjdk11_bisheng_extended.system_x86-64_linux #301](https://ci.adoptium.net/job/Test_openjdk11_bisheng_extended.system_x86-64_linux/301/)
17:13:45  Starting building: [Test_openjdk11_bisheng_sanity.perf_x86-64_linux #301](https://ci.adoptium.net/job/Test_openjdk11_bisheng_sanity.perf_x86-64_linux/301/)
17:13:45  Starting building: [Test_openjdk11_bisheng_sanity.functional_x86-64_linux #301](https://ci.adoptium.net/job/Test_openjdk11_bisheng_sanity.functional_x86-64_linux/301/)
17:13:45  Starting building: [Test_openjdk11_bisheng_sanity.external_x86-64_linux #71](https://ci.adoptium.net/job/Test_openjdk11_bisheng_sanity.external_x86-64_linux/71/)
17:13:45  Starting building: [Test_openjdk11_bisheng_extended.functional_x86-64_linux #301](https://ci.adoptium.net/job/Test_openjdk11_bisheng_extended.functional_x86-64_linux/301/)
17:13:45  Starting building: [Test_openjdk11_bisheng_extended.openjdk_x86-64_linux #98](https://ci.adoptium.net/job/Test_openjdk11_bisheng_extended.openjdk_x86-64_linux/98/)
17:13:45  Starting building: [Test_openjdk11_bisheng_extended.perf_x86-64_linux #98](https://ci.adoptium.net/job/Test_openjdk11_bisheng_extended.perf_x86-64_linux/98/)
17:13:45  Starting building: [Test_openjdk11_bisheng_special.functional_x86-64_linux #98](https://ci.adoptium.net/job/Test_openjdk11_bisheng_special.functional_x86-64_linux/98/)
17:13:48  Starting building: [Test_Job_Auto_Gen #1501](https://ci.adoptium.net/job/Test_Job_Auto_Gen/1501/)
17:13:49  Starting building: [Test_Job_Auto_Gen #1503](https://ci.adoptium.net/job/Test_Job_Auto_Gen/1503/)
17:13:49  Starting building: [Test_Job_Auto_Gen #1502](https://ci.adoptium.net/job/Test_Job_Auto_Gen/1502/)
17:15:43  Build [Test_openjdk11_bisheng_sanity.external_x86-64_linux #71](https://ci.adoptium.net/job/Test_openjdk11_bisheng_sanity.external_x86-64_linux/71/) completed: FAILURE
```
Would that indicate that the autogen jobs are potentially getting run in parallel with the executions so the regen may not have completed before the jobs were run? I'm not sure that would fully cover why we're still seeing

**Comment by sxa** [comment]
> The regen didn't happen as the typo. The fix PR https://github.com/adoptium/ci-jenkins-pipelines/pull/907

Approved, although it looks like quite a few regen jobs did get triggered during the weekly pipeline earlier this week despite this.

**Comment by sxa** [comment]
> EDIT: After regeneration the sanity.openjdk config.xml for Bisheng increased in size from 33879 bytes to 45691 so it looks like there was quite a bit missing. The list at the top of https://ci.adoptium.net/job/Test_Job_Auto_Gen/ is notable by the absence of some of the other variants having been included for some reason.

@sophia-guo is there an easy way to force regeneration of some of these? It's not quite clear to me why only some of them are getting regenerated. It looks like me manually regenerating Test_openjdk11_bisheng_sanity.openjdk_x86-64_linux` has made a difference though, so that's good - I'd just rather not have to run that for all of the problematic test jobs if possible.

**Comment by sophia-guo** [comment]
> EDIT: Looks like the three generator jobs only covered special.openjdk dev.functional and dev.openjdk

Those regens are triggerred as there are no jobs generated before. If you see the regened job https://ci.adoptium.net/job/Test_openjdk11_bisheng_special.openjdk_x86-64_linux/ there is only one build right now.

> I'm running generators on [sanity.openjdk](https://ci.adoptium.net/job/Test_Job_Auto_Gen/1510/) and [extended.openjdk](https://ci.adoptium.net/job/Test_Job_Auto_Gen/1511/)

Those jobs were generated before so won't be regened by default unless set AQA_AUTO_GEN=true. With Weekly regen, they will be regened weekly.

**Comment by sxa** [comment]
> > I'm running generators on [sanity.openjdk](https://ci.adoptium.net/job/Test_Job_Auto_Gen/1510/) and [extended.openjdk](https://ci.adoptium.net/job/Test_Job_Auto_Gen/1511/)
> 
> Those jobs were generated before so won't be regened by default unless set AQA_AUTO_GEN=true. With Weekly regen, they will be regened weekly.

Hmmm so can you tell why the Bisheng11 sanity.openjdk was not not regenerated before https://ci.adoptium.net/view/Test_openjdk/job/Test_openjdk11_bisheng_sanity.openjdk_x86-64_linux/301 which was started from the weekly pipeline at https://ci.adoptium.net/job/build-scripts/job/weekly-openjdk11-pipeline/156/ a few days ago? It looks like the `openjdk11-pipeline` from the weekly job didn't get  `AQA_AUTO_GEN=true` but from what you've said it should have if I understand you correctly. I'm probably missing something really obvious here :-)

**Comment by sxa** [comment]
The only ones appearing more than once in the latest log are:
```
    190 Skipped parameter `ADDITIONAL_ARTIFACTS_REQUIRED` as it is undefined on `Test_openjdk21_hs_extended.openjdk_x86-64_linux_testList_0` (#38).
    264 Skipped parameter `ADDITIONAL_ARTIFACTS_REQUIRED` as it is undefined on `Test_openjdk21_hs_extended.openjdk_x86-64_linux_testList_1` (#38).
    190 Skipped parameter `ADDITIONAL_ARTIFACTS_REQUIRED` as it is undefined on `Test_openjdk21_hs_extended.openjdk_x86-64_linux_testList_2` (#38).
    190 Skipped parameter `BASE_DOCKER_REGISTRY_CREDENTIAL_ID` as it is undefined on `Test_openjdk21_hs_extended.openjdk_x86-64_linux_testList_0` (#38).
    264 Skipped parameter `BASE_DOCKER_REGISTRY_CREDENTIAL_ID` as it is undefined on `Test_openjdk21_hs_extended.openjdk_x86-64_linux_testList_1` (#38).
    190 Skipped parameter `BASE_DOCKER_REGISTRY_CREDENTIAL_ID` as it is undefined on `Test_openjdk21_hs_extended.openjdk_x86-64_linux_testList_2` (#38).
      2 Skipped parameter `BASE_DOCKER_REGISTRY_CREDENTIAL_ID` as it is undefined on `Test_openjdk22_hs_extended.openjdk_s390x_linux_testList_0` (#13).
      2 Skipped parameter `BASE_DOCKER_REGISTRY_CREDENTIAL_ID` as it is undefined on `Test_openjdk22_hs_extended.openjdk_s390x_linux_testList_1` (#13).
      2 Skipped parameter `BASE_DOCKER_REGISTRY_CREDENTIAL_ID` as it is undefined on `Test_openjdk22_hs_extended.openjdk_s390x_linux_testList_2` (#13).
     31 Skipped parameter `RERUN_ITERATIONS` as it is undefined on `Test_openjdk17_hs_sanity.external_x86-64_alpine-linux` (#10).
```
I'm tempted to just remove the testList ones and let them be regenerated when required. It looks like the last one referencing `RERUN_ITERATIONS` has not yet be regenerated since that update was made.

**Comment by sxa** [comment]
> Weekly pipeline with parameter releaseType=Weekly will trigger the regen https://ci.adoptium.net/job/build-scripts/job/weekly-openjdk11-pipeline/156/parameters/. The code is in https://github.com/adoptium/ci-jenkins-pipelines/pull/883/files.
> 
> The regen didn't happen as the typo. The fix PR [adoptium/ci-jenkins-pipelines#907](https://github.com/adoptium/ci-jenkins-pipelines/pull/907)

@sophia-guo I'm seeing a lot of the `testList_` jobs showing this warning (including hotspot ones) - should those be regenerated along with the main jobs, and if not can we force a regen of those? If not would it be a good idea to just delete them all since they will be regenerated oe would that cause problems?

Also Test_Job_Auto_Gen seems to have the issue for the `TEST_JOB_NAME` and `LIGHT_WEIGHT_CHECKOUT` parameters so that will require some manual intervention.

**Comment by sophia-guo** [comment]
There are two other types of test jobs naming as `_testList_` and `_rerun`. Those jobs are generated if no jobs exist or if test job parameter `GENERATE_JOBS` is true, which current can't be set by upstream weekly pipeline unless we add extra parameter.

**Comment by sxa** [comment]
From discussion with @smlambert in one of our calls today it is safe to delete all the `testList` jobs so I will start pruning ✂️

**Comment by smlambert** [comment]
I am not aware that we have or support a PIPELINE_DISPLAY_NAME for those test jobs.  Also, as noted in today's team call, I do not think that there has been a recent jdk17 triggered run, last one was 4 days ago.

**Comment by sophia-guo** [comment]
`PIPELINE_DISPLAY_NAME` was set for `AQA_Test_Pipeline`, which should not affect test jobs generated or triggered by non-`AQA_Test_Pipeline`. That is saying those warning message should not be in adoptium jenkins server log? It might be in  in Eclipse temurin jenkins server log. As [consume parameters dynamically in AQA Test Pipeline](https://github.com/adoptium/aqa-tests/pull/5048) parameter [PIPELINE_DISPLAY_NAME](https://github.com/adoptium/aqa-tests/blob/master/buildenv/jenkins/aqaTestPipeline.groovy#L19) will pass to children test jobs, which will not fail tests but feel unnecessary. Might just not pass the parameter `PIPELINE_DISPLAY_NAME` to children jobs like some other parameters   https://github.com/adoptium/aqa-tests/blob/master/buildenv/jenkins/aqaTestPipeline.groovy#L112-L114 @llxia

**Comment by sxa** [comment]
> That is saying those warning message should not be in adoptium jenkins server log? It might be in in Eclipse temurin jenkins server log.

Just to clarify, I'm only looking at logs for ci.adoptium.net so everything I'm posting is related to that (Is that what you mean by "adoptium jenkins" as distinct from "Eclipse temurin jenkins" which is the Temurin Compliance one?

**Comment by sxa** [comment]
> The latest weekly jdk17 run was on Feb 4th https://ci.adoptium.net/job/build-scripts/job/weekly-openjdk17-pipeline/158/. The regen was set by weekly_release_pipeline.groovy. I don't think tagged builds will do the regen.

OK then I think we need to look at setting it for the regular pipelines triggered by the beta triggers, since the weekly ones are irrelevant now with the tag-triggered jobs running all of the tests that we had in the weekly jobs.

**Comment by sophia-guo** [comment]
Yes, that's what confused me (PIPELINE_DISPLAY_NAME warning in  ci.adoptium.net  logs. )

**Comment by sophia-guo** [comment]
> OK then I think we need to look at setting it for the regular pipelines triggered by the beta triggers, since the weekly ones are irrelevant now with the tag-triggered jobs running all of the tests that we had in the weekly jobs.

That might also mean alway regen the test jobs for pipeline - not sure is it necessary?

**Comment by sxa** [comment]
> > OK then I think we need to look at setting it for the regular pipelines triggered by the beta triggers,

> That might also mean alway regen the test jobs for pipeline - not sure is it necessary?

I'm happy to take advice from you on any solution that resolves the warnings as that is the goal here :-) I was told previously that the regen was the primary way to prevent these from occuring when changes to the parameters occurred so that seemed like one suitable solution.

**Comment by sxa** [comment]
Noting that I triggered a reload of the configuration this morning since the logs were still showing issues in some of the testList jobs which were no longer there after yesterday's cleanup. I'll keep an eye on it for the next day and see which ones (if any) continue to show up.

**Comment by llxia** [comment]
PIPELINE_DISPLAY_NAME was added via https://github.com/adoptium/aqa-tests/pull/4186. It is for displaying info in the AQA Test Pipeline job only. It does not affect any child jobs, but it does not need to pass into child jobs. We should skip it.

**Comment by llxia** [comment]
Issue opened: https://github.com/adoptium/aqa-tests/issues/5089

**Comment by sxa** [comment]
# Summary of today (27 Feb 2024)

### Build jobs (FYI @andrew-m-leonard):

<details>
<summary>NODE_LABEL issues on some build jobs</summary>
      1 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk11u/jdk11u-aix-ppc64-hotspot` (#1,077).
      1 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk8u/jdk8u-aix-ppc64-hotspot` (#1,178).
      1 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk8u/jdk8u-linux-aarch64-hotspot` (#1,075).
      1 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk8u/jdk8u-mac-x64-hotspot` (#1,214).
      1 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk/jdk-linux-riscv64-temurin` (#128).
      1 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk/jdk-mac-aarch64-hotspot` (#39).
</details>

<details>
<summary>openjdk-build-pr-tester jobs</summary>
      1 Skipped parameter `ghprbActualCommit` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `ghprbActualCommitAuthor` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `ghprbActualCommitAuthorEmail` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `ghprbAuthorRepoGitUrl` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `ghprbCommentBody` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `ghprbCredentialsId` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `ghprbGhRepository` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `ghprbPullAuthorEmail` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `ghprbPullAuthorLogin` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `ghprbPullAuthorLoginMention` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `ghprbPullDescription` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `ghprbPullId` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `ghprbPullLink` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `ghprbPullLongDescription` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `ghprbPullTitle` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `ghprbSourceBranch` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `ghprbTargetBranch` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `ghprbTriggerAuthor` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `ghprbTriggerAuthorEmail` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `ghprbTriggerAuthorLogin` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `ghprbTriggerAuthorLoginMention` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `GIT_BRANCH` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
      1 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk/jdk-linux-riscv64-temurin` (#128).
      1 Skipped parameter `sha1` as it is undefined on `build-scripts-pr-tester/openjdk-build-pr-tester` (#1,859).
</details>


### Test jobs:

<details>
<summary>hotspot external tests (are they disabled so not regenerated?)</summary>
      1 Skipped parameter `RELEASE_TAG` as it is undefined on `Test_openjdk11_hs_extended.external_ppc64le_linux_quarkus_openshift` (#27).
      1 Skipped parameter `RELEASE_TAG` as it is undefined on `Test_openjdk11_hs_extended.external_ppc64le_linux_tomcat` (#27).
      1 Skipped parameter `RELEASE_TAG` as it is undefined on `Test_openjdk11_hs_sanity.external_x86-64_linux_functional-test` (#117).
      1 Skipped parameter `RELEASE_TAG` as it is undefined on `Test_openjdk8_hs_extended.external_ppc64le_linux_quarkus` (#49).
      1 Skipped parameter `RELEASE_TAG` as it is undefined on `Test_openjdk8_hs_extended.external_x86-64_linux_elasticsearch` (#27).
      1 Skipped parameter `RELEASE_TAG` as it is undefined on `Test_openjdk8_hs_sanity.external_s390x_linux_quarkus` (#44).
      1 Skipped parameter `RUNTIME_NAME` as it is undefined on `Test_openjdk11_hs_extended.external_ppc64le_linux_quarkus_openshift` (#27).
      1 Skipped parameter `RUNTIME_NAME` as it is undefined on `Test_openjdk11_hs_extended.external_ppc64le_linux_tomcat` (#27).
      1 Skipped parameter `RUNTIME_NAME` as it is undefined on `Test_openjdk11_hs_sanity.external_x86-64_linux_functional-test` (#117).
      1 Skipped parameter `RUNTIME_NAME` as it is undefined on `Test_openjdk8_hs_extended.external_ppc64le_linux_quarkus` (#49).
      1 Skipped parameter `RUNTIME_NAME` as it is undefined on `Test_openjdk8_hs_extended.external_x86-64_linux_elasticsearch` (#27).
      1 Skipped parameter `RUNTIME_NAME` as it is undefined on `Test_openjdk8_hs_sanity.external_s390x_linux_quarkus` (#44).
      1 Skipped parameter `TKG_BRANCH` as it is undefined on `Test_openjdk11_hs_extended.external_ppc64le_linux_quarkus_openshift` (#27).
      1 Skipped parameter `TKG_BRANCH` as it is undefined on `Test_openjdk11_hs_extended.external_ppc64le_linux_tomcat` (#27).
      1 Skipped parameter `TKG_BRANCH` as it is undefined on `Test_openjdk11_hs_sanity.external_x86-64_linux_functional-test` (#117).
      1 Skipped parameter `TKG_BRANCH` as it is undefined on `Test_openjdk8_hs_extended.external_ppc64le_linux_quarkus` (#49).
      1 Skipped parameter `TKG_BRANCH` as it is undefined on `Test_openjdk8_hs_extended.external_x86-64_linux_elasticsearch` (#27).
      1 Skipped parameter `TKG_BRANCH` as it is undefined on `Test_openjdk8_hs_sanity.external_s390x_linux_quarkus` (#44).
      1 Skipped parameter `TKG_REPO` as it is undefined on `Test_openjdk11_hs_extended.external_ppc64le_linux_quarkus_openshift` (#27).
      1 Skipped parameter `TKG_REPO` as it is undefined on `Test_openjdk11_hs_extended.external_ppc64le_linux_tomcat` (#27).
      1 Skipped parameter `TKG_REPO` as it is undefined on `Test_openjdk11_hs_sanity.external_x86-64_linux_functional-test` (#117).
      1 Skipped parameter `TKG_REPO` as it is undefined on `Test_openjdk8_hs_extended.external_ppc64le_linux_quarkus` (#49).
      1 Skipped parameter `TKG_REPO` as it is undefined on `Test_openjdk8_hs_extended.external_x86-64_linux_elasticsearch` (#27).
      1 Skipped parameter `TKG_REPO` as it is undefined on `Test_openjdk8_hs_sanity.external_s390x_linux_quarkus` (#44).
</details>

<details>
<summary> j9 jobs (mostly external ones)</summary>
      1 Skipped parameter `DYNAMIC_COMPILE` as it is undefined on `Test_openjdk11_j9_sanity.perf_ppc64_aix` (#310).
      1 Skipped parameter `DYNAMIC_COMPILE` as it is undefined on `Test_openjdk17_j9_extended.openjdk_s390x_linux` (#69).
      1 Skipped parameter `DYNAMIC_COMPILE` as it is undefined on `Test_openjdk17_j9_extended.perf_aarch64_linux` (#69).
      1 Skipped parameter `DYNAMIC_COMPILE` as it is undefined on `Test_openjdk17_j9_extended.system_aarch64_linux` (#201).
      1 Skipped parameter `DYNAMIC_COMPILE` as it is undefined on `Test_openjdk17_j9_sanity.perf_x86-64_windows` (#235).
      1 Skipped parameter `RELEASE_TAG` as it is undefined on `Test_openjdk11_j9_extended.external_ppc64le_linux_jacoco` (#4).
      1 Skipped parameter `RELEASE_TAG` as it is undefined on `Test_openjdk11_j9_extended.external_ppc64le_linux_wildfly` (#25).
      1 Skipped parameter `RELEASE_TAG` as it is undefined on `Test_openjdk11_j9_extended.external_x86-64_linux_elasticsearch` (#34).
      1 Skipped parameter `RELEASE_TAG` as it is undefined on `Test_openjdk11_j9_extended.external_x86-64_linux_quarkus_openshift` (#30).
      1 Skipped parameter `RELEASE_TAG` as it is undefined on `Test_openjdk11_j9_extended.external_x86-64_linux_wycheproof` (#33).
      1 Skipped parameter `RELEASE_TAG` as it is undefined on `Test_openjdk11_j9_sanity.system_x86-64_mac` (#31).
      1 Skipped parameter `RELEASE_TAG` as it is undefined on `Test_openjdk17_j9_extended.perf_x86-64_linux_xl` (#5).
      1 Skipped parameter `RELEASE_TAG` as it is undefined on `Test_openjdk17_j9_sanity.perf_aarch64_linux_xl` (#26).
      1 Skipped parameter `RELEASE_TAG` as it is undefined on `Test_openjdk8_j9_extended.external_x86-64_linux_thorntail-mp-tck` (#26).
      1 Skipped parameter `RUNTIME_NAME` as it is undefined on `Test_openjdk11_j9_extended.external_ppc64le_linux_jacoco` (#4).
      1 Skipped parameter `RUNTIME_NAME` as it is undefined on `Test_openjdk11_j9_extended.external_ppc64le_linux_wildfly` (#25).
      1 Skipped parameter `RUNTIME_NAME` as it is undefined on `Test_openjdk11_j9_extended.external_x86-64_linux_elasticsearch` (#34).
      1 Skipped parameter `RUNTIME_NAME` as it is undefined on `Test_openjdk11_j9_extended.external_x86-64_linux_quarkus_openshift` (#30).
      1 Skipped parameter `RUNTIME_NAME` as it is undefined on `Test_openjdk11_j9_extended.external_x86-64_linux_wycheproof` (#33).
      1 Skipped parameter `RUNTIME_NAME` as it is undefined on `Test_openjdk8_j9_extended.external_x86-64_linux_thorntail-mp-tck` (#26).
      1 Skipped parameter `TKG_BRANCH` as it is undefined on `Test_openjdk11_j9_extended.external_ppc64le_linux_jacoco` (#4).
      1 Skipped parameter `TKG_BRANCH` as it is undefined on `Test_openjdk11_j9_extended.external_ppc64le_linux_wildfly` (#25).
      1 Skipped parameter `TKG_BRANCH` as it is undefined on `Test_openjdk11_j9_extended.external_x86-64_linux_elasticsearch` (#34).
      1 Skipped parameter `TKG_BRANCH` as it is undefined on `Test_openjdk11_j9_extended.external_x86-64_linux_quarkus_openshift` (#30).
      1 Skipped parameter `TKG_BRANCH` as it is undefined on `Test_openjdk11_j9_extended.external_x86-64_linux_wycheproof` (#33).
      1 Skipped parameter `TKG_BRANCH` as it is undefined on `Test_openjdk8_j9_extended.external_x86-64_linux_thorntail-mp-tck` (#26).
      1 Skipped parameter `TKG_REPO` as it is undefined on `Test_openjdk11_j9_extended.external_ppc64le_linux_jacoco` (#4).
      1 Skipped parameter `TKG_REPO` as it is undefined on `Test_openjdk11_j9_extended.external_ppc64le_linux_wildfly` (#25).
      1 Skipped parameter `TKG_REPO` as it is undefined on `Test_openjdk11_j9_extended.external_x86-64_linux_elasticsearch` (#34).
      1 Skipped parameter `TKG_REPO` as it is undefined on `Test_openjdk11_j9_extended.external_x86-64_linux_quarkus_openshift` (#30).
      1 Skipped parameter `TKG_REPO` as it is undefined on `Test_openjdk11_j9_extended.external_x86-64_linux_wycheproof` (#33).
      1 Skipped parameter `TKG_REPO` as it is undefined on `Test_openjdk8_j9_extended.external_x86-64_linux_thorntail-mp-tck` (#26).
      1 Skipped parameter `USE_TESTENV_PROPERTIES` as it is undefined on `Test_openjdk11_j9_sanity.perf_ppc64_aix` (#310).
      1 Skipped parameter `USE_TESTENV_PROPERTIES` as it is undefined on `Test_openjdk17_j9_extended.openjdk_s390x_linux` (#69).
      1 Skipped parameter `USE_TESTENV_PROPERTIES` as it is undefined on `Test_openjdk17_j9_extended.perf_aarch64_linux` (#69).
      1 Skipped parameter `USE_TESTENV_PROPERTIES` as it is undefined on `Test_openjdk17_j9_extended.system_aarch64_linux` (#201).
      1 Skipped parameter `USE_TESTENV_PROPERTIES` as it is undefined on `Test_openjdk17_j9_sanity.perf_x86-64_windows` (#235).
</details>


<details>
<summary> Other tests (mostly Grinder or non-hotspot)</summary>
     1 Skipped parameter `ARCHIVE_TEST_RESULTS` as it is undefined on `grinder_sandbox_new_testList_0` (#26).
      2 Skipped parameter `DYNAMIC_COMPILE` as it is undefined on `Test_openjdk11_bisheng_sanity.perf_x86-64_linux` (#304).
      1 Skipped parameter `DYNAMIC_COMPILE` as it is undefined on `Test_openjdk11_bisheng_sanity.system_aarch64_linux` (#313).
      1 Skipped parameter `DYNAMIC_COMPILE` as it is undefined on `Test_openjdk11_dragonwell_special.functional_x86-64_windows` (#87).
      1 Skipped parameter `GENERATE_JOBS` as it is undefined on `Test_openjdk11_dragonwell_special.functional_x86-64_windows` (#87).
      1 Skipped parameter `NON_AQA_TEST_REPOS` as it is undefined on `Grinder_rerun` (#10).
      1 Skipped parameter `NON_AQA_TEST_REPOS` as it is undefined on `Grinder_testList_3` (#3).
      1 Skipped parameter `NON_AQA_TEST_REPOS` as it is undefined on `Grinder_testList_4` (#2).
      1 Skipped parameter `NON_AQA_TEST_REPOS` as it is undefined on `Test_openjdk11_bisheng_sanity.openjdk_aarch64_linux_rerun` (#1).
      1 Skipped parameter `PLATFORM_AND_MACHINE` as it is undefined on `Grinder_rerun` (#10).
      1 Skipped parameter `PLATFORM_AND_MACHINE` as it is undefined on `Grinder_testList_3` (#3).
      1 Skipped parameter `PLATFORM_AND_MACHINE` as it is undefined on `Grinder_testList_4` (#2).
      1 Skipped parameter `PLATFORM_AND_MACHINE` as it is undefined on `Test_openjdk11_bisheng_sanity.openjdk_aarch64_linux_rerun` (#1).
      1 Skipped parameter `RELEASE_TAG` as it is undefined on `Test_openjdk17_hs_sanity.openjdk_arm64_mac` (#24).
      1 Skipped parameter `RELEASE_TAG` as it is undefined on `Test_openjdk17_hs_sanity.perf_arm64_mac` (#24).
      1 Skipped parameter `RERUN_ITERATIONS` as it is undefined on `Grinder_testList_4` (#2).
      2 Skipped parameter `RERUN_ITERATIONS` as it is undefined on `Test_openjdk11_bisheng_sanity.perf_x86-64_linux` (#304).
      1 Skipped parameter `RERUN_ITERATIONS` as it is undefined on `Test_openjdk11_fast_startup_extended.perf_x86-64_linux` (#72).
      1 Skipped parameter `TEST_IMAGES_REQUIRED` as it is undefined on `Grinder_rerun` (#10).
      1 Skipped parameter `TEST_IMAGES_REQUIRED` as it is undefined on `Grinder_testList_3` (#3).
      1 Skipped parameter `TEST_IMAGES_REQUIRED` as it is undefined on `Grinder_testList_4` (#2).
      1 Skipped parameter `TKG_SHA` as it is undefined on `Test_openjdk11_bisheng_sanity.openjdk_aarch64_linux_rerun` (#1).
      2 Skipped parameter `USE_TESTENV_PROPERTIES` as it is undefined on `Test_openjdk11_bisheng_sanity.perf_x86-64_linux` (#304).
      1 Skipped parameter `USE_TESTENV_PROPERTIES` as it is undefined on `Test_openjdk11_bisheng_sanity.system_aarch64_linux` (#313).
      1 Skipped parameter `USE_TESTENV_PROPERTIES` as it is undefined on `Test_openjdk11_dragonwell_special.functional_x86-64_windows` (#87).
</details>

**Comment by sxa** [comment]
@sophia-guo Is there a way to trigger regeneration of some of the jobs that aren't run regularly e.g. the external ones, and the non-hotspot ones like j9? Those seem to be most of the ones in the test area that are remaining.

**Comment by sxa** [comment]
The build ones that are left seem to be `hotspot` ones, or the pr tester (Mostly `NODE_LABEL`)

**Comment by sxa** [comment]
> > @sophia-guo Is there a way to trigger regeneration of some of the jobs that aren't run regularly e.g. the external ones, and the non-hotspot ones like j9? Those seem to be most of the ones in the test area that are remaining.
> 
> Did you mean the not running jobs ( not triggered to run or disabled ) are still generating warning messages?

It seems that way, yes.

**Comment by sxa** [comment]
> @sophia-guo Is there a way to trigger regeneration of some of the jobs that aren't run regularly e.g. the external ones, and the non-hotspot ones like j9? Those seem to be most of the ones in the test area that are remaining.

Can someone confirm the above to me please? If there is an easy way to trigger this that doesn't involve running a full openjdkXX-pipeline with the aqaAutoGen flag? This seems like an easy fix if I can just regen the out of date pipelines.

**Comment by sxa** [comment]
Getting on top of these now I think - there were over 5000 yesterday, but as of half way through today we've only had 8. I'll do another summary over the weekend. We could do with regenerating the `-hotspot` build jobs I think (especially since we seem to have [switched all of them off](https://github.com/adoptium/ci-jenkins-pipelines/issues/987) for now which I would suggest wasn't ideal unless they're running outside the triggers) .

@andrew-m-leonard is there a way to point the regen jobs at a configuration that includes the non-temurin ones?

There have also been instances of these on the pr-tester jobs which could probably do with a bit of maintenance (although they're not showing up in the log today so far)

**Comment by andrew-m-leonard** [comment]
So I think the other variants need to be in the pipelines targetConfiguration to be re-gen'd so If you could point the regen at a fork with such a change you could do possibly?

**Comment by sxa** [comment]
Yeah that's what I wanted to do but it wasn't clear how to make that work. Is there a way to point it at an alternate location or would I have to create a new branch, put the changes in, edit the generator job (always a risk!) to use the fork, and then run it ... Or is there a simpler way?

There is DEFAULTS_URL which can point externally (although the default in the description of that points to something at the old AdoptOpenJDK repo, but I don't think the equavalint file in the new repo is the one I want)

**Comment by sxa** [comment]
From today and yesterday:
<details>
<summary>Build jobs</summary>

```
      1 Skipped parameter `DYNAMIC_COMPILE` as it is undefined on `build-scripts/jobs/jdk8u/jdk8u-linux-x64-hotspot_SmokeTests` (#146).
      1 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk11u/jdk11u-alpine-linux-aarch64-temurin` (#43).
      1 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk17u/jdk17u-linux-s390x-hotspot` (#50).
      1 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk8u/jdk8u-alpine-linux-aarch64-temurin` (#140).
      1 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk8u/jdk8u-linux-arm-hotspot` (#830).
      1 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk8u/jdk8u-linux-s390x-temurin` (#272).
      1 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk8u/jdk8u-solaris-sparcv9-hotspot` (#644).
      1 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk8u/jdk8u-windows-x86-32-hotspot` (#1,112).
      1 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk/jdk-mac-aarch64-hotspot` (#36).
      1 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk/jdk-mac-aarch64-hotspot` (#37).
      1 Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts/jobs/jdk/jdk-mac-aarch64-hotspot` (#38).
```
</details>

<details>
<summary>Test jobs (Mostly JDK11)</summary>

```
      1 Skipped parameter `ACTIVE_NODE_TIMEOUT` as it is undefined on `Test_openjdk11_dragonwell_sanity.openjdk_x86-64_linux` (#459).
      1 Skipped parameter `ACTIVE_NODE_TIMEOUT` as it is undefined on `Test_openjdk11_j9_extended.openjdk_ppc64le_linux_xl` (#7).
      1 Skipped parameter `ACTIVE_NODE_TIMEOUT` as it is undefined on `Test_openjdk11_j9_extended.openjdk_s390x_linux_xl` (#9).
      1 Skipped parameter `ADOPTOPENJDK_SYSTEMTEST_BRANCH` as it is undefined on `grinder_sandbox_iteration_2` (#18).
      1 Skipped parameter `ADOPTOPENJDK_SYSTEMTEST_REPO` as it is undefined on `grinder_sandbox_iteration_2` (#18).
      1 Skipped parameter `ARCHIVE_TEST_RESULTS` as it is undefined on `grinder_sandbox_iteration_2` (#18).
      1 Skipped parameter `DEBUG_IMAGES_REQUIRED` as it is undefined on `grinder_sandbox_iteration_2` (#18).
      1 Skipped parameter `DYNAMIC_COMPILE` as it is undefined on `Test_openjdk11_dragonwell_sanity.openjdk_x86-64_linux` (#459).
      1 Skipped parameter `DYNAMIC_COMPILE` as it is undefined on `Test_openjdk11_j9_sanity.openjdk_s390x_linux` (#674).
      1 Skipped parameter `DYNAMIC_COMPILE` as it is undefined on `Test_openjdk11_j9_sanity.openjdk_x86-64_windows` (#708).
      1 Skipped parameter `GENERATE_JOBS` as it is undefined on `Test_openjdk11_dragonwell_sanity.openjdk_x86-64_linux` (#459).
      1 Skipped parameter `GITHUB_TOKEN` as it is undefined on `refactor_openjdk_release_tool_new` (#25).
      1 Skipped parameter `JDK_BRANCH` as it is undefined on `Test_openjdk11_dragonwell_sanity.openjdk_x86-64_linux` (#459).
      1 Skipped parameter `JDK_BRANCH` as it is undefined on `Test_openjdk11_j9_extended.openjdk_ppc64le_linux_xl` (#7).
      1 Skipped parameter `JDK_BRANCH` as it is undefined on `Test_openjdk11_j9_extended.openjdk_s390x_linux_xl` (#9).
      1 Skipped parameter `JDK_REPO` as it is undefined on `Test_openjdk11_dragonwell_sanity.openjdk_x86-64_linux` (#459).
      1 Skipped parameter `JDK_REPO` as it is undefined on `Test_openjdk11_j9_extended.openjdk_ppc64le_linux_xl` (#7).
      1 Skipped parameter `JDK_REPO` as it is undefined on `Test_openjdk11_j9_extended.openjdk_s390x_linux_xl` (#9).
      1 Skipped parameter `LABEL_ADDITION` as it is undefined on `Test_openjdk11_j9_extended.openjdk_ppc64le_linux_xl` (#7).
      1 Skipped parameter `LABEL_ADDITION` as it is undefined on `Test_openjdk11_j9_extended.openjdk_s390x_linux_xl` (#9).
      1 Skipped parameter `NON_AQA_TEST_REPOS` as it is undefined on `Grinder_security` (#1).
      1 Skipped parameter `NON_AQA_TEST_REPOS` as it is undefined on `Test_openjdk11_bisheng_extended.openjdk_x86-64_linux_rerun` (#1).
      1 Skipped parameter `NUM_MACHINES` as it is undefined on `Test_openjdk11_dragonwell_sanity.openjdk_x86-64_linux` (#459).
      1 Skipped parameter `OPENJ9_SYSTEMTEST_BRANCH` as it is undefined on `grinder_sandbox_iteration_2` (#18).
      1 Skipped parameter `OPENJ9_SYSTEMTEST_REPO` as it is undefined on `grinder_sandbox_iteration_2` (#18).
      1 Skipped parameter `PARALLEL` as it is undefined on `Test_openjdk11_dragonwell_sanity.openjdk_x86-64_linux` (#459).
      1 Skipped parameter `PLATFORM_AND_MACHINE` as it is undefined on `Grinder_security` (#1).
      1 Skipped parameter `PLATFORM_AND_MACHINE` as it is undefined on `Test_openjdk11_bisheng_extended.openjdk_x86-64_linux_rerun` (#1).
      1 Skipped parameter `RELEASE_TAG` as it is undefined on `grinder_sandbox_iteration_2` (#18).
      1 Skipped parameter `RELEASE_TAG` as it is undefined on `Test_openjdk11_hs_sanity.external_x86-64_linux_openliberty-mp-tck` (#497).
      1 Skipped parameter `RELEASE_TAG` as it is undefined on `Test_openjdk11_j9_extended.external_ppc64le_linux_quarkus_openshift` (#25).
      1 Skipped parameter `RERUN_ITERATIONS` as it is undefined on `Test_openjdk11_dragonwell_sanity.openjdk_x86-64_linux` (#459).
      1 Skipped parameter `RUNTIME_NAME` as it is undefined on `grinder_sandbox_iteration_2` (#18).
      1 Skipped parameter `RUNTIME_NAME` as it is undefined on `Test_openjdk11_hs_sanity.external_x86-64_linux_openliberty-mp-tck` (#497).
      1 Skipped parameter `RUNTIME_NAME` as it is undefined on `Test_openjdk11_j9_extended.external_ppc64le_linux_quarkus_openshift` (#25).
      1 Skipped parameter `STF_BRANCH` as it is undefined on `grinder_sandbox_iteration_2` (#18).
      1 Skipped parameter `STF_REPO` as it is undefined on `grinder_sandbox_iteration_2` (#18).
      1 Skipped parameter `TEST_IMAGES_REQUIRED` as it is undefined on `grinder_sandbox_iteration_2` (#18).
      1 Skipped parameter `TKG_BRANCH` as it is undefined on `Test_openjdk11_hs_sanity.external_x86-64_linux_openliberty-mp-tck` (#497).
      1 Skipped parameter `TKG_BRANCH` as it is undefined on `Test_openjdk11_j9_extended.external_ppc64le_linux_quarkus_openshift` (#25).
      1 Skipped parameter `TKG_REPO` as it is undefined on `Test_openjdk11_hs_sanity.external_x86-64_linux_openliberty-mp-tck` (#497).
      1 Skipped parameter `TKG_REPO` as it is undefined on `Test_openjdk11_j9_extended.external_ppc64le_linux_quarkus_openshift` (#25).
      1 Skipped parameter `TKG_SHA` as it is undefined on `Grinder_scala` (#16).
      1 Skipped parameter `TKG_SHA` as it is undefined on `Grinder_security` (#1).
      1 Skipped parameter `TKG_SHA` as it is undefined on `Test_openjdk11_bisheng_extended.openjdk_x86-64_linux_rerun` (#1).
      1 Skipped parameter `USE_TESTENV_PROPERTIES` as it is undefined on `Grinder_scala` (#16).
      1 Skipped parameter `USE_TESTENV_PROPERTIES` as it is undefined on `Test_openjdk11_dragonwell_sanity.openjdk_x86-64_linux` (#459).
      1 Skipped parameter `USE_TESTENV_PROPERTIES` as it is undefined on `Test_openjdk11_j9_sanity.openjdk_s390x_linux` (#674).
      1 Skipped parameter `USE_TESTENV_PROPERTIES` as it is undefined on `Test_openjdk11_j9_sanity.openjdk_x86-64_windows` (#708).
```
</details>

**Comment by sxa** [comment]
The test autogen jobs seem to have two parameters causing problems too:
- `2024-05-03 12:37:35.013+0000 [id=4318864]	WARNING	hudson.model.ParametersAction#filter: Skipped parameter `LIGHT_WEIGHT_CHECKOUT` as it is undefined on `Test_Job_Auto_Gen` (#5,090). Set `-Dhudson.model.ParametersAction.keepUndefinedParameters=true` to allow undefined parameters to be injected as environment variables or `-Dhudson.model.ParametersAction.safeParameters=[comma-separated list]` to whitelist specific parameter names, even though it represents a security breach or `-Dhudson.model.ParametersAction.keepUndefinedParameters=false` to no longer show this message.`
- `2024-05-03 12:37:35.014+0000 [id=4318864]	WARNING	hudson.model.ParametersAction#filter: Skipped parameter `TEST_JOB_NAME` as it is undefined on `Test_Job_Auto_Gen` (#5,090). Set `-Dhudson.model.ParametersAction.keepUndefinedParameters=true` to allow undefined parameters to be injected as environment variables or `-Dhudson.model.ParametersAction.safeParameters=[comma-separated list]` to whitelist specific parameter names, even though it represe`

**Comment by sxa** [comment]
@smlambert @andrew-m-leonard Is there a way we can regen the jobs from the [earlier comment](https://github.com/adoptium/infrastructure/issues/2774#issuecomment-2042567798) (Plus the pr-tester ones which can show up problems)>
I'm not clear on how to do this, so if you're able to trigger it and put the information in here that would be appreciated so we can clear off the relatively small number of remaining ones that we seem to have.

**Comment by andrew-m-leonard** [comment]
@sxa this look like job instances ? eg.build-scripts/jobs/jdk/jdk-mac-aarch64-hotspot` (#38)
Can't we just delete those particular job numbers?

**Comment by sxa** [comment]
> @sxa this look like job instances ? eg.build-scripts/jobs/jdk/jdk-mac-aarch64-hotspot` (#38) Can't we just delete those particular job numbers?

A fair comment - I hadn't actually spotted that it was on non-current ones. If you confirm you're ok with us just deleting them I'm happy to go through and do that as a background task.

That would likely just leave things like this on the PR tester jobs:
`/var/log/jenkins/jenkins.log.1:2024-05-02 08:36:36.633+0000 [id=4239618]	WARNING	hudson.model.ParametersAction#filter: Skipped parameter `NODE_LABEL` as it is undefined on `build-scripts-pr-tester/build-test/jobs/jdk/jdk-alpine-linux-x64-temurin` (#9). Set `-Dhudson.model.ParametersAction.keepUndefinedParameters=true` to allow undefined parameters to be injected as environment variables or `-Dhudson.model.ParametersAction.safeParameters=[comma-separated list]` to whitelist specific parameter names, even though it represents a security breach or `-Dhudson.model.ParametersAction.keepUndefinedParameters=false` to no longer show this message.`

**Comment by sxa** [comment]
> Test jobs (28) J9(16)/Dragonwell(9)/Bisheng(3)

From a comment from Shelley in a meeting earlier today it seems likely that running https://ci.eclipse.org/temurin-compliance/job/AQA_Test_Pipeline with a `VARIANT` set appropriately and `AUTO_AQA_GEN` checked

(Although I'm not sure if this will regen the `_quarkus_openshift`, `_xl` and `_rerun` jobs so it would be good to have a way of doing that)

Similarly I'm unclear if we have a way to regen the various Grinder variants (although almost all of those in the twisty earlier were `grinder_sandbox_iteration_2`

**Comment by sxa** [comment]
Noting also that we're getting a couple of the parameters giving  warnings on `Test_Job_Auto_Gen` as called from `AQA_Test_Pipeline` which will likely require some sort of manual remediation:
- 2024-05-07 16:43:03.074+0000 [id=85459]	WARNING	hudson.model.ParametersAction#filter: Skipped parameter `LIGHT_WEIGHT_CHECKOUT` as it is undefined on `Test_Job_Auto_Gen` (#5,239). Set `-Dhudson.model.ParametersAction.keepUndefinedParameters=true` to allow undefined parameters to be injected as environment variables or `-Dhudson.model.ParametersAction.safeParameters=[comma-separated list]` to whitelist specific parameter names, even though it represents a security breach or `-Dhudson.model.ParametersAction.keepUndefinedParameters=false` to no longer show this message.
- 2024-05-07 16:43:03.076+0000 [id=85459]	WARNING	hudson.model.ParametersAction#filter: Skipped parameter `TEST_JOB_NAME` as it is undefined on `Test_Job_Auto_Gen` (#5,239). Set `-Dhudson.model.ParametersAction.keepUndefinedParameters=true` to allow undefined parameters to be injected as environment variables or `-Dhudson.model.ParametersAction.safeParameters=[comma-separated list]` to whitelist specific parameter names, even though it represents a security breach or `-Dhudson.model.ParametersAction.keepUndefinedParameters=false` to no longer show this message.

**Comment by smlambert** [comment]
FYI, TEST_JOB_NAME only applies to AQA_Test_Pipeline, there is no parameter of that name used in the generated jobs (nor do we want their to be).

**Comment by smlambert** [comment]
> Similarly I'm unclear if we have a way to regen the various Grinder variants (although almost all of those in the twisty earlier were grinder_sandbox_iteration_2

I have been slowly removing grinder variants especially if they were last run 1 yr or more ago

**Comment by sxa** [comment]
> FYI, TEST_JOB_NAME only applies to AQA_Test_Pipeline, there is no parameter of that name used in the generated jobs (nor do we want their to be).

Makes sense, although the warning would suggest that `AQA_Test_pipeline` is trying to pass that parametr down to `Test_Job_Auto_Gen` somwhere.

**Comment by smlambert** [comment]
> Makes sense, although the warning would suggest that AQA_Test_pipeline is trying to pass that parametr down to Test_Job_Auto_Gen somwhere.

The code cycles through all parameters and passes them down (with some exceptions at https://github.com/adoptium/aqa-tests/blob/master/buildenv/jenkins/aqaTestPipeline.groovy#L110-L114).  It can be added to the set of exceptions that are not added to `childParams` so that when the `downstreamJob` is created (at https://github.com/adoptium/aqa-tests/blob/master/buildenv/jenkins/aqaTestPipeline.groovy#L147), the content of childParams is without it.

**Comment by sxa** [comment]
Deep dive into one of them (Corretto JDK8 sanity/system) for reference. There are only 13 entries relating to test skipped parameters today so far, and 8 of them are from different parameters on this job so yesterday's regens appear to have made quite a big difference.
- Job 480 is giving the skipped message with `JDK_REPO` `JDK_BRANCH` `PARALLEL` `NUM_MACHINES` `USE_TESTENV_PROPERTIES` `GENERATE_JOBS` `ACTIVE_NODE_TIMEOUT` and `DYNAMIC_COMPILE`
- Regenerating at https://ci.adoptium.net/job/Test_Job_Auto_Gen/5362/
- There is no difference in the individual builds (despite the reference to build 480 in the log). Diff between old and new `config.xml` for the job is here in case it's useful for future historic reference (it's quite big since it has reference to adoptopenjdk repository - the last run of that job was April 14th 2022): 
[2774.Test_openjdk8_corretto_sanity.system_x86-64_linux.diff.txt](https://github.com/adoptium/infrastructure/files/15247976/2774.Test_openjdk8_corretto_sanity.system_x86-64_linux.diff.txt)

**Comment by sxa** [comment]
> I have been slowly removing grinder variants especially if they were last run 1 yr or more ago

@smlambert Can "suffixed" jobs like https://ci.adoptium.net/job/Test_openjdk11_hs_sanity.external_s390x_linux_system-test/ be removed too? The ones I've looked at today seem to be ones that haven't been run in nearly 3 years so I guess they were likely generated as some sort of experiment and are no longer required.

I don't want to be deleting them myself though - would rather someone from the test side with knowledge of the particular jobs that should be in scope was able to handle it.

**Comment by smlambert** [comment]
I have a script that will look for jobs that have not been run in XX number of months, and optionally delete them if 'deleteJobs' parameter equals true.  I think we should run such a job occasionally on the server to cull old, not-used jobs.  I will do a pass in the coming weeks.

**Comment by sxa** [comment]
There are a lot of new instances of this showing up relating to the new `RERUN_FAILURE` parameter. I think we're going to need some sort of process for ensuring that all the pipelines are suitable updated when we make changes to the parameters.

**Comment by sxa** [comment]
Also @steelhead31 There are a few relating to `create_installer_linux`:
- `UPSTREAM_JOB_NAME`
- `UPSTREAM_JOB_NUMBER`
- `UPLOAD_TESTRESULTS_ONLY`
- `TIMESTAMP`
- `GITHUB_TOKEN`
- `ARTIFACTS_TO_SKIP`

**Comment by sxa** [comment]
> This should be resolved now, let me know if not.

Do you have a link to the PR where you resolved this, as it may be useful in the future to others resolving similar problems elsewhere :-)
